scripts/lib/CIME/XML/entry_id.py

```diff
@@ -319,19 +319,19 @@ def _get_value(self, node, attribute=None, resolved=True, subgroup=None):
 
     def get_child_content(self, vid, childname):
         val = None
-        node = self.get_optional_node("entry", {"id" : vid})
+        node = self.get_optional_child("entry", {"id" : vid})
         if node is not None:
             val = self.get_element_text(childname, root=node)
         return val
 
     def get_elements_from_child_content(self, childname, childcontent):
-        nodes = self.get_nodes("entry")
+        nodes = self.get_children("entry")
         elements = []
         for node in nodes:
             content = self.get_element_text(childname, root=node)
-            expect(content is not None,"No childname {} for id {}".format(childname, node.get("id")))
+            expect(content is not None,"No childname {} for id {}".format(childname, self.get(node, "id")))
             if content == childcontent:
-                elements.append(deepcopy(node))
+                elements.append(node)
```

Comment:
why don't we need the deepcopy any longer?  I remember this causing problems because without deepcopy you got a pointer to the original instead of a copy.

Reply:
@jedwards4b , I double-checked and add_elements_by_group was the only caller of this method and it was already making a deepcopy of these nodes.